### PR TITLE
chore(data-warehouse): Tree UI for data warehouse

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- too many tabs

## Changes

- reduce tabs and use a tree viewer

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/lib/components/DatabaseTableTree/Tree.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.Tree {
+    background: var(--bg-light);
+}
+
+.Tree__root {
+    padding: 1rem;
+}
```
Comment (Gilbert09) [inline frontend/src/lib/components/DatabaseTableTree/Tree.scss:7]: Can we use tailwind for this?

File: frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    item: TreeItemLeaf
+    depth: number
+    onClick?: (row: DataWarehouseSceneRow) => void
+    selected?: boolean
+}
+
+export function TreeRow({ item, onClick, selected }: TreeRowProps): JSX.Element {
+    const _onClick = (): void => {
+        onClick && onClick(item.table)
+    }
```
Comment (Gilbert09) [inline frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx:21]: Probably worth wrapping this in a `useCallback`

File: frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    selectedRow?: DataWarehouseSceneRow | null
+}
+
+export function TreeFolderRow({ item, depth, onClick, selectedRow }: TreeFolderRowProps): JSX.Element {
+    const [collapsed, setCollapsed] = useState(false)
+    const { name, items, emptyLabel } = item
+
+    const _onClick = (): void => {
+        setCollapsed(!collapsed)
+    }
```
Comment (Gilbert09) [inline frontend/src/lib/components/DatabaseTableTree/TreeRow.tsx:46]: Same here with `useCallback`

File: frontend/src/scenes/data-warehouse/external/DataWarehouseExternalScene.tsx [hunk 4]
```diff
@@ -61,21 +174,73 @@ export function DataWarehouseExternalScene(): JSX.Element {
...
+                            {selectedRow.external_data_source ? (
+                                <></>
+                            ) : (
+                                selectedRow.url_pattern && (
+                                    <>
+                                        <span className="card-secondary mt-2">Files URL pattern</span>
+                                        <span>{selectedRow.url_pattern}</span>
+                                    </>
+                                )
+                            )}
```
Comment (Gilbert09) [inline frontend/src/scenes/data-warehouse/external/DataWarehouseExternalScene.tsx:218]: We can probably remove the ternary and do `{!selectedRow.external_data_source && selectedRow.url_pattern && <>...</>}`

File: frontend/src/layout/navigation-3000/sidepanel/panels/activity/notificationsLogic.tsx [hunk 5]
```diff
@@ -0,0 +1,245 @@
+import { actions, events, kea, listeners, path, reducers, selectors } from 'kea'
```
Comment (Gilbert09) [inline frontend/src/layout/navigation-3000/sidepanel/panels/activity/notificationsLogic.tsx:1]: Is this file meant to be in this PR? I don't see it referenced from any of the data warehouse stuff.

**Review by Gilbert09 — APPROVED** [review]
A few comments - nothing blocking though, and UX feels good for right now.

File: frontend/src/layout/navigation-3000/sidepanel/panels/activity/notificationsLogic.tsx (hunk 5 above)
Comment (EDsCODE) [reply frontend/src/layout/navigation-3000/sidepanel/panels/activity/notificationsLogic.tsx]: bad merge!

**Review by Gilbert09 — APPROVED** [review]

**Pull request merged**